### ISSUE TITLE
MotorStorm [BCES00006]: Cars stutter when using PPU LLVM

### ISSUE THREAD

**Issue by legend800** [RPCS3/rpcs3#8376]
Cars stutter/jump constantly. Present on user vehicle and on other racers. Not present on bikes.

[RPCS3-MS.log.gz](https://github.com/RPCS3/rpcs3/files/4741229/RPCS3-MS.log.gz)

Config:
Windows 10 64-bit
GeForce GTX 1080
i7-6700k @ 4GHZ
16GB RAM

**Comment by dio-gh** [comment]
Is the crowd T-posing also only present on PPU LLVM? Perhaps the ticket could be generalized then to "MotorStorm issues with PPU LLVM" or similar.

**Comment by legend800** [comment]
I'm not sure what t-posing you're talking about - never seen the issue. The npcs are frozen but not in tpose on both interp and LLVM. You'll have to test yourself.

![2020-06-06 18_08_21-FPS_ 30 01 _ Vulkan _ 0 0 10-d9d4a22b-rsx_volatile _ MotorStorm™  BCES00006](https://user-images.githubusercontent.com/20983844/83957982-8b23fc80-a821-11ea-9763-9019f5a8f3bf.png)

**Comment by dio-gh** [comment]
Huh, I went ahead and scanned through every mention of MotorStorm I saw on Discord and YouTube and indeed couldn't find the issue. I could swear I saw it, but I can't find it again. Maybe the NPCs being still tripped me off. Luckily them being still is accurate behaviour though.

**Comment by legend800** [comment]
Still active as of 0.0.16-12427.

**Comment by segbed** [comment]
still have the same issue "Cars stutter when using PPU LLVM" aka "vehicles are warping forward/backward". It can be partially fixed by downclocking the CPU (e.g. 10600k at 60% via power manager) but for sure it is just a workaround.
Is there any chance it will be fixed this year?
also mentioned here: https://forums.rpcs3.net/thread-183124-page-2.html

**Comment by segbed** [comment]
some of the latest few builds partially or fully fixed the issue.
But game itself are freezing after 1 min of gameplay and has a lot of graphic glitches (maybe even more than before). But imho it is great progress, maybe soon it will be fully playable.

**Comment by legend800** [comment]
This issue has not improved at all unfortunately. I've been testing this issue for years so no idea what you're basing that on. It's a logic issue - once it's fixed, it'll be gone completely. This isn't an issue that will just get a little better here and there. 

And also the gfx have regressed badly with flickering black dirt all over the place now. So no progress just regression. I hope they can focus on getting this fan favorite console exclusive game working once and for all soon. 

![2021-11-01 23_27_38-Window](https://user-images.githubusercontent.com/20983844/139797071-d943edb3-f1d5-4ba0-a3c1-605a9528d7de.png)

**Comment by MSuih** [comment]
Non-transparent particles issue is tracked in #10936

**Comment by Megamouse** [comment]
I've seen this game run quite well lately. Is the stutter still an issue ?

**Comment by legend800** [comment]
No change here. Cars still stutter as much as they always have for years. First level doesn't have the issue as much as every other level, especially lvl 3 Speed Kills. Only Intepreter fixes it but runs at 15fps max on a high end machine.

**Comment by Juansero29** [comment]
Yep, also experiencing this. Any way we can help solving this issue?

**Comment by Juansero29** [comment]
I'm cross referencing here some forum posts wich address this issue with different "possible fixes" which seem to be not definitive and don't work for everyone.

These include
- running game on linux
- running game at 50% cpu process speed adjusting it in windows power mode window
- some other settings in the simulator which seem to help the issue for some people.



Forum threads: 

- https://forums.rpcs3.net/thread-183124-page-2.html?highlight=Motorstorm
- https://forums.rpcs3.net/thread-204386.html?highlight=Motorstorm
https://forums.rpcs3.net/thread-183124-page-2.html?highlight=Motorstorm

I advice people who want to help solve this in the first place to condense all the informatiom about fixes (rumors or verified info) in this issue, and maybe also try out different settings combinations, platforms and OS and write out your experience in here so that together we can find a solution. Hope this helps finding a way to get the best motorstorm experience possible.

**Comment by legend800** [comment]
The best config I've found is SPU = Interpreter and SPU threads = 1. Fixes 95% of the stuttering with full fps BUT they audio is garbled. I care more about video and can just play some audio in the background.

![2022-12-27 09_05_52-Settings](https://user-images.githubusercontent.com/20983844/209698750-7e8c62ae-f744-413e-aee6-45b4b531468d.png)

Using PPU = Interpreter also fixes it completely but you get half speed.

**Comment by eapiova** [comment]
Tested both on Windows 11 and on Arch Linux
Can confirm that this bug only happens on Windows on my rig, while on Linux is perfectly fine.

Also SPU = Interpreter and SPU threads = 1 doesn't do the trick, while PPU = Interpreter fixes it with very slow speed

Config:
Windows 11 64-bit (vs. Arch Linux)
GeForce GTX 4070 TI
AMD R5 5600x
32GB RAM

**Comment by segbed** [comment]
also confirmed that it is still no progress with fixing this issue. I've played with all possible settings without success. 
I will open a bottle of wine when I will be able to play this game on PC with no issues. Hopefully this year.

**Comment by Juansero29** [comment]
Just here again to add that it has been a few months since I play the game using Linux Mint and there's a lot less flickering. It's not completely gone, but it's still 80% better than any Windows experience so far. So for hardcore fans out there, definitely install Linux and play the game there, it's a blast :)

**Comment by Valtekken** [comment]
Can confirm this issue is happening to me as well. It isn't unbearable and the game is still playable, but it happens often enough to be noticeable. Windows 11, Ryzen 5600X, Radeon RX 6600, 16GB RAM.

**Comment by Mario-119** [comment]
As of Build 0.0.32-16861 this issue is still present under Windows

Specs: 
Windows 11
Ryzen 5 7600X
Radeon RX 6750 XT
32GB-DDR5

**Comment by AnubyteCode** [comment]
Nothing tested fixes this issue without making the game slow or creating other bugs.

win 11 24h2 26100.1742
i9 11900h
rtx 3060 6gb
64gb ddr4

**Comment by pkrol1218** [comment]
Sam's issues with cars on win 11
Amd 5600x 
Pny 1070 
16 ddr4, was 32 😞
X570 wifi

**Comment by AnubyteCode** [comment]
Ok, so I have almost completely eliminated stutter. It's entirely tollerable, reduced probably 95%.

Patches
- disable dynamic res
- disable motion blue
- unlock fps

Settings not default:
- SPU block size, Mega
- write/read color buffers
- accurate RSX reservation
- accurate SPU DMA
- vulkan queue scheduler, fast
- RSX FIFO, atomic
- driver wake-up delay, 100 microseconds

I am currently working on a heavily customized profile that has eliminated the car stutter to just 2% of default, but still needs performance optimizations and unnecessary settings cleaned, using the EU version of the game.

_Personal verdict: Mostly playable to playable._

**Comment by AnubyteCode** [comment]
Here's my current settings from the NA and EU versions. Make of them what you will, trial them, laugh even. Seems like changing any one uncommon setting breaks the effect of another so I simply don't have the time to repeat for days. They work with my system (info above a few posts). 

Timer Accuracy to "All Timers" helped quite a bit, cutting the jitter in half (after adding it to the wiki defaults). Just about anything accuracy related was required, except xfloat and vulcan queue.  Both files work for their respective versions, but may be compatible (EU file still has my scaling settings). EU game version is 3.1, I don't know what if any changes were made to the engine over NA's 1.0 but it generally plays better.

[config_BCUS98137.zip](https://github.com/user-attachments/files/17792876/config_BCUS98137.zip)
[config_BCES00006.zip](https://github.com/user-attachments/files/17792877/config_BCES00006.zip)

**Comment by alexconsole04** [comment]
While testing, I only used Accurate Cache Line Store and it helped a lot with the physics, which I was more concerned about.
It still stutters, but so far, cars don't bounce as much, more accurate to the game on PS3.

[RPCS3.log.gz](https://github.com/user-attachments/files/20840437/RPCS3.log.gz)

**Comment by jchabin** [comment]
As of Windows builds being unblocked, this is completely fixed for me. Tested using both the recommended settings on the Wiki, and with resolution scaling + 60FPS, no more car stuttering.

https://www.youtube.com/watch?v=N90SrfOT8jM

My only disappointment is that I had noticed issues similar to this one in both LittleBigPlanet 3 and LittleBigPlanet Karting, neither of which are fixed yet.

**Comment by Valtekken** [comment]
> As of Windows builds being unblocked, this is completely fixed for me. Tested using both the recommended settings on the Wiki, and with resolution scaling + 60FPS, no more car stuttering.
> 
> https://www.youtube.com/watch?v=N90SrfOT8jM
> 
> My only disappointment is that I had noticed issues similar to this one in both LittleBigPlanet 3 and LittleBigPlanet Karting, neither of which are fixed yet.

It's fixed? Oh thank God, I can actually finish the game now. 
The cars stuttering like that made it hard to keep playing

**Comment by alexconsole04** [comment]
Running a few races, haven't gotten any stutter whatsoever. Using only wiki settings, no patches were used.

[RPCS3.log.gz](https://github.com/user-attachments/files/22450306/RPCS3.log.gz)

<img width="1280" height="720" alt="Image" src="https://github.com/user-attachments/assets/f7112325-496e-4179-a503-ed832e41eaaf" />
Hooray, this means that the game can finally be moved to playable

**Comment by ajbonner** [comment]
So excited will test this today as well!

**Comment by Mario-119** [comment]
Thrilled that this got around to getting patched

**Comment by segbed** [comment]
wow it is incredibly cool!!! tested, run smooth as butter in ~90FPS. Thanks to devs!!!

**Comment by ajbonner** [comment]
Confirmed this works great 60fps at 2k for me which, but most importantly stutter free. Thank you so much to all the developers involved!

**Comment by hoholee12** [comment]
although the player car does not stutter anymore, opponent cars still stutter(flicker). its not completely fixed...

**Comment by digant73** [comment]
@hoholee12 open a new issue on that, possibly providing a video etc.

**Comment by alexconsole04** [comment]
Haven't seen any of the other cars stutter. Not sure what's going on your end.
